app/src/main/java/com/example/immadisairaj/codeforces/SubmissionAdapter.java

```diff
@@ -100,8 +101,10 @@ public void onClick(View v) {
                 if (submissionList.get(position).getProblem().getProblemsetName() == null) {
                     if (submissionList.get(position).getProblem().getContestId() >= 100000)
                         url += "gym/" + submissionList.get(position).getProblem().getContestId() + "/problem/" + submissionList.get(position).getProblem().getIndex();
-                    else
-                        url += "problemset/problem/" + submissionList.get(position).getProblem().getContestId() + "/" + submissionList.get(position).getProblem().getIndex();
+                    else {
```

Comment:
can there be a dialog box where the user can either select the problem question or the submission s/he made